x-pack/plugins/lens/public/indexpattern_datasource/indexpattern.tsx

```diff
@@ -320,6 +326,34 @@ export function getIndexPatternDatasource({
     getDropProps,
     onDrop,
 
+    getCustomWorkspaceRenderer: (state: IndexPatternPrivateState, dragging: DraggingIdentifier) => {
+      if (dragging.field === undefined || dragging.indexPatternId === undefined) {
+        return undefined;
+      }
+
+      const draggedField = dragging as DraggingIdentifier & {
```

Comment:
It seems like we need to do the check whether the action is actually available here as well, right?

Reply:
We do not need to check if the visualize geo field action is available anywhere else because geo_point and geo_shape fields will not exist in the list when visualize geo field action is not available. Therefore, the field check will never return true since geo fields will have been removed prior to this logic.